### DISCUSSION TITLE
credential error

### DISCUSSION THREAD

**Discussion by XAVIERXEXT** [community/community#53397] [Pull Requests]
### Select Topic Area

Show & Tell

### Body

when pushing code to the GitHub from Xcode using terminal it's asking username and password , and even if I am providing correct username and password or token it's showing error as below.   ===>>  
 remote: Permission to surajbadaik/Demo.git denied to surajbadaik.
fatal: unable to access 'https://github.com/surajbadaik/Demo.git/': The requested URL returned error: 403

**Comment by airtower-luna** [comment]
Your password definitely won't work for Git operations, a token should. When creating the token make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).

If it doesn't work anyway there's probably some issue with the token not being entered correctly (usually you should copy & paste), but that's hard to verify from the outside. :sweat_smile:
